src/factories/offsets.ts

```diff
@@ -12,24 +11,24 @@ type RemoveOffsetParameters = {
   nodeId: string,
 }
 
+export type Offsets = Awaited<ReturnType<typeof offsetsFactory>>
+export type OffsetFactory = Offsets['factory']
+
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function offsetsFactory() {
-  const offsets: Offsets = new Map()
+export async function offsetsFactory() {
+  const config = await waitForConfig()
+  const offsets: Map<number, Map<string, number> | undefined> = new Map()
 
   function getOffset(axis: number): number {
-    const values = offsets.get(axis)
-
-    if (!values) {
-      return 0
-    }
+    const values = offsets.get(axis) ?? []
 
-    return Math.max(...values.values(), 0)
+    return Math.max(...values.values(), config.styles.nodeHeight)
```

Comment:
This is evolving a bit to factor in the entire row height and not just the "offset" the name of this utility is likely to change to reflect but waiting to see how this shakes out